src/telemetry.ts

```diff
@@ -129,7 +139,15 @@ export class BatchTelemetryReporter {
 
     private send(): void {
         for (const eventName in this._eventBuckets) {
-            const bucket = this._eventBuckets[eventName];
+            let bucket = this._eventBuckets[eventName];
+            const eventConfig = this._eventConfig[eventName];
+            if (eventConfig) {
+                const diff = bucket.length - eventConfig.batchCap;
+                if (diff > 0) {
+                    bucket = bucket.slice(0, eventConfig.batchCap);
+                    logger.log(`Truncate ${diff} ${eventName} event entries.`);
```

Comment:
If we ever drop events we'll probably want to know how many total events we got, how many were successful, and the total/average time they took to execute

Reply:
It can be done. But it's out of the scope of this PR. Let's figure out and reach to an agreement on what is needed to be done on the client side and understand the issues if we don't have this data, then decide if it's absolutely necessary to take that into regard.

Why do you think the dropped data needed to be taken into regard? How would it affect the insight we are trying to deduce?

Reply:
I think at the very least, we should have some counter of how many we are dropping.  Averages and other aggregation we can add later.

Reply:
What I'm hoping to avoid is having some notifications with errors, or some notifications that are taking a very long time to get processed, and not getting that information because we drop those events.

Reply:
I can add a counter for how many we dropped.

@digeff From a statistical point of view, I incline to assume that the remaining part follows the same distribution as the rest. In other words, notification with error and long processing has equal opportunity to appear in the retained portion.